oak-query-spi/src/main/java/org/apache/jackrabbit/oak/spi/query/QueryIndex.java

```diff
@@ -491,6 +510,17 @@ public IndexPlan build() {
                     private final boolean deprecated =
                             Builder.this.deprecated;
                     private final boolean logWarningForPathFilterMismatch = Builder.this.logWarningForPathFilterMismatch;
+                    private final Map<Level, List<String>> additionalMessages = Builder.this.additionalMessages;
+
+                    private String getAdditionalMessageString() {
+                        StringBuilder stringBuilder = new StringBuilder();
+                        for (Map.Entry<Level, List<String>> messages : additionalMessages.entrySet()) {
+                            stringBuilder.append(messages.getKey()).append(" : ").append(messages.getValue()).append(", ");
```

Comment:
Fabrizios approach is fine... if you want to use a loop, it would have to be like this (I didn't test):

    StringBuilder buff = new StringBuilder();
    for (Map.Entry<Level, List<String>> messages : additionalMessages.entrySet()) {
        if (!buff.isEmpty()) {
            buff.append(", ");
        }
        buff.append(messages.getKey() + " : " + messages.getValue());
    }
    return buff.toString();

I'm not saying it's better that Fabrizios approach... just that this would be the "common" approach when using a loop.

Reply:
Fixed, thanks!